contracts/test/Spender.t.sol

```diff
@@ -202,6 +220,23 @@ contract SpenderTest is Test {
         spender.spendFromUser(user, address(lon), 100);
     }
 
+    function testCannotSpendFromUserWithNoReturnValueToken() public {
+        uint256 userBalance = noReturnERC20.balanceOf(user);
+        vm.expectRevert("Spender: ERC20 transferFrom failed");
+        spender.spendFromUser(user, address(noReturnERC20), userBalance + 1);
```

Comment:
Why setting transfer amount to `userBalance + 1`?

Reply:
To cause the transaction to revert so Spender can catch it:
`require(callSucceed, "Spender: ERC20 transferFrom failed");`

Reply:
Maybe we can one more test case which succeeds in transferring a `noReturnERC20` token.

Reply:
Ok, I think it's about the naming. `testCannotSpendFromUserWithNoReturnValueToken` looks like spender cannot spend token without return value to me. Perhaps these can be merged as one test case `testCannotSpendFromUserInsufficientBalance` and test special tokens in `testSpendFromUser` as well.

Reply:
Good point. Will rename and add the tests. Though I will prefer separating them into two cases: `testCannotSpendFromUserInsufficientBalance_NoReturnValueToken`, `testCannotSpendFromUserInsufficientBalance_NoRevertToken` and `testSpendFromUserWithNoReturnValueToken`.